### ISSUE TITLE
pdk validate should have an option to limit validation to a single module

### ISSUE THREAD

**Issue by jcbollinger** [puppetlabs/pdk#476]
I have a development workspace containing a complete clone of my site configuration, including multiple modules, many of them third-party.  In that environment, I would like to be able to `pdk validate` a single module at a time -- whichever I happen to be working on.  PDK does not seem to provide a way to do that.

Instead, when I run `pdk validate` from a module's root directory, I often see validation diagnostics for other modules present in my environment.  PDK seems to be either using module metadata and / or discovering and recursively following references to other modules' classes and defined types; I'm not sure which.  I assume this is intentional, but it is counter-productive for me, in part because it makes validation runs take significantly longer, but more importantly because the diagnostics I want to see are hard to discriminate from those I don't care about.

There are several ways a UI for narrowing validation scope might work:
 - Ideal for me would be for PDK to simply restrict its scope to validating files in the sub-tree rooted at its working directory.  It might refer to files outside that tree to _inform_ some of its validations, but it would not recursively validate files outside the tree.
 - As a variation on the above, when run anywhere within a module's directory tree, PDK might validate only the contents of that module, as if by stepping backward up the tree to the module's root, and then operating as described above.
 - It would also be ok with me if I had to specify a non-default option to get one of the above behaviors.  For example, I might have to say '`pdk validate --local`'.  No sweat, as long as it doesn't take an inordinate amount of extra typing.
 - Or PDK might instead provide a `--module` or `--modules` parameter by which I could designate an exclusive list of modules to validate.  This one is about at the limit of my willingness to type.

Those are roughly in order of my personal preference, and not all mutually exclusive.

Note: I see issue #329, which is different, but in many ways complementary.  It may be possible to devise a unified solution addressing both this issue and that.

**Comment by scotje** [comment]
@jcbollinger can you elaborate a bit on the layout of your workspace?

I just tried creating an example workspace that looks like this:

```
.
├── metadata.json
├── modules
└── site
    ├── test_site_bad
    └── test_site_mod
```

And in that workspace, I'm able to `cd site/test_site_mod` and successfully run `pdk validate` and have it validate only that module. Is it possible there are symlinks or something between the modules that are causing PDK to traverse to adjacent modules?

**Comment by jcbollinger** [comment]
Hi @scotje 

Thanks for having a look at this.  I updated to PDK 1.10 since reporting the issue, and I am now having trouble reproducing the problem.  I am uncertain at this point whether updating PDK solved the issue by itself.  Frankly, I got out of the habit of routinely running `pdk validate`, since it was a pain, so I'm not sure when things changed.

In any case, I am pretty confident that the situation that inspired the report involved modules related _at least_ through references in their `metadata.json` files.  I do also have some symlinking going on in `spec/fixtures/modules` for modules that reference other local, unpublished modules, and some of my tests use the `compile.with_all_deps` matcher.  I thought at the time that one or more of these forms of module association were probably involved in the crossover I observed.
